myapp/src/components/DisplayMap.js

```diff
@@ -10,13 +10,35 @@ import LocationPin  from '../images/marker-icon.png';
 import Select from 'ol/interaction/Select';
 import Overlay from 'ol/Overlay';
 import * as ReactDOM from 'react-dom';
-import './DisplayMap.css';
+import { makeStyles } from '@material-ui/core';
+
+const useStyles = makeStyles(theme => ({
+  popupContainerStyle: {
+    width: '100%',
+    height: '600px',
+    position: 'relative',
+  },
+  popupContentStyle: {
+    backgroundColor: 'white',
+    border: '1px solid #ccc',
+    padding: '10px',
+    boxShadow: '0px 2px 6px rgba(0, 0, 0, 0.1)',
+    maxWidth: '300px',
+  },
+  strongStyle: {
+    display: 'block',
+    fontWeight: 'bold',
+    marginBottom: '8px',
+  },
+}))
 
 const DisplayMap = ({ tasks }) => {
+  const classes = useStyles();
+
   const mapRef = useRef(null);
   const layerRef = useRef(null);
   const mapInstance = useRef(null);
```

Comment:
Why you need mapRef and mapInstance ?

Reply:
When I first created the map , it was rendering twice or even more , so I created a mapInstance which is a reference to the map itself and makes sure it displays only one time during the initialization , and the mapRef is assigned to the target where i render the map with new changes